app/src/main/java/com/example/tictactoe/MainActivity.kt

```diff
@@ -43,4 +48,30 @@ class MainActivity : AppCompatActivity() {
     override fun onSupportNavigateUp(): Boolean {
         return navController.navigateUp() || super.onSupportNavigateUp()
     }
+
+    // The following two methods can only be called from GameFragment via WinningDialog
+    override fun onWinningDialogPositiveClick(
+        dialog: DialogFragment,
+        viewModel: TicTacToeViewModel,
+        fragment: GameFragment
+    ) {
+        // Play again with the same names
+        viewModel.resetGameState()
+        // Refresh fragment from scratch so the UI reflects the cleared data
+        fragment.parentFragmentManager.beginTransaction()
+            .detach(fragment)
```

Comment:
could a detach / attach just be replaced with a `replace` call?

Reply:
I'll switch it out and give it a go! Thanks!